source/core/index.ts

```diff
@@ -1009,8 +1009,10 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 
 					return result;
 				}) as typeof http.request,
-				cache as CacheableRequest.StorageAdapter,
-			));
+				cache as StorageAdapter,
+			).createCacheableRequest();
```

Comment:
This pattern looks very weird. Why are you calling `createCacheableRequest` if there is already `new CacheableRequest`?

Reply:
@szmarczak In an earlier version of `cacheable-request` the class constructor was returning the method but we can't return that in `typescript`. `createCacheableRequest ` returns a function.

Reply:
I see. In this case, it should be created automatically and exposed via e.g. `cacheable.request()`.

Reply:
I'm good with the current solution for now

Reply:
@alphmth since we are doing a major change and they are willing to do this change should we go with what @szmarczak is suggesting as it is more elegant.

Reply:
@jaredwray sounds good I will make change

Reply:
@szmarczak - change is now made and updated to do `request()` instead of `createCacheableRequest()`